File: cpp/BubbleSort.cpp
#include<iostream>

// function
void BubbleSort(int a[], int n){
    // outer for loop runs n-1 times in worst case
    for(int i= 0; i<n-1; i++){
        // take flag to see weather given element is sorted or not
        int flag = 0;
        for(int j=0; j<n-i-1; j++){
            if(a[j]>a[j+1]){
                int temp = a[j];
                a[j]= a[j+ 1];
                a[j+1]=temp;
                // set flag to 1, means to given arrays is not sorted, so swaping happens
                flag =1;
            }
        }
        if(flag==0)
            break;
    }
}

int main(){
    int n;
    int a[10];
    std::cout<<"enter the number of elements: ";
    std::cin>>n;
    std::cout<<"enter any " <<n<<" elements: ";
    
    for(int i=0; i<n; i++)
    std::cin>>a[i];
    //int a[]={3,10,5,3,2,1};
    BubbleSort(a,n);
    
    std::cout<<"Sorted arrays are: ";
    for(int i=0; i<n; i++)
    std::cout<<a[i]<<" ";
    return 0;
}

File: java/src/BubbleSort.java
class BubbleSort{
    public static void main (String[] args) {
        /* code */
        int[] a = {5,2,10,0,1};
        SortBuble(a, 5);
        System.out.println("Sorted arrays are: ");
        for(int i=0; i<5; i++)
        System.out.print(a[i]+ " ");
        System.out.println();
    }// main
    
    // Add you comment here
    
    
    // sorting function
    public static void SortBuble(int a[], int n){
        for(int i= 0; i<n-1; i++){
            for(int j=0; j<n-i-1; j++){
                if(a[j]>a[j+1]){
                    int temp = a[j];
                    a[j]= a[j+ 1];
                    a[j+1]=temp;
                }
            }
        }
    }// SortBuble

}// BubbleSort